### PR TITLE
[CR] [WIP] Changes sights to mount to rails, not weapons.

### PR DESCRIPTION
**Pull request by hyper2snyper**
```SUMMARY: Content "Changes sights to attach to rails not weapons"```
I had a thought of making sights attach to rails, not weapon types. Really, i just want an ACOG on my handgun. So ill be using this to learn a bit of C++ and to mess with the code a bit.

### CODE REVIEW

File: data/json/items/gunmod/rail.json [hunk 1]
```diff
@@ -46,6 +46,23 @@
     "min_skills": [ [ "gun", 1 ] ],
     "add_mod": [ [ "sights", 1 ] ]
   },
+  {
+    "id": "pistol_sight_rail",
+    "type": "GUNMOD",
+    "name": "pistol rail mount",
+    "description": "A rail mount for using larger sites.",
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/rail.json]:
```suggestion
    "description": "A rail mount for using larger sights.",
```

File: data/json/items/gunmod/sights.json [hunk 2]
```diff
@@ -139,13 +139,32 @@
...
     "sight_dispersion": 0,
     "aim_speed": 0,
     "min_skills": [ [ "weapon", 2 ], [ "gun", 4 ] ],
     "flags": [ "DISABLE_SIGHTS", "ZOOM" ]
   },
+  {
+    "id": "sniper_scope",
+    "type": "GUNMOD",
+    "name": "sniper's scope",
+    "description": "A very large sight, this scope has a range of nearly 4000",
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/sights.json]: 4000 of what? Also, period in the end.

File: data/json/items/gunmod/rail.json [hunk 3]
```diff
@@ -64,6 +82,60 @@
     "aim_speed": 8,
     "min_skills": [ [ "weapon", 2 ], [ "gun", 1 ] ]
   },
+  {
+    "id": "small_rail",
+    "type": "GUNMOD",
+    "name": "small sight rail",
+    "description": "A small rail, for attaching scopes and the like. This one can only fit smaller sights."
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/rail.json]:
```suggestion
    "description": "A small rail, for attaching scopes and the like.  This one can only fit smaller sights."
```

File: data/json/items/gunmod/rail.json [hunk 4]
```diff
@@ -64,6 +82,60 @@
...
+    "mod_targets": [ "pistol" ],
+    "min_skills": [ [ "gun", 1 ] ],
+    "add_mod": [ [ "sights", 1 ] ],
+    "flags": [ "DISABLE_SIGHTS" ]
+  },
+  {
+    "id": "med_rail",
+    "type": "GUNMOD",
+    "name": "medium sight rail",
+    "description": "A medium sized rail, for attaching scopes and the like. This one fits medium sized sights."
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/rail.json]:
```suggestion
    "description": "A medium sized rail, for attaching scopes and the like.  This one fits medium sized sights."
```

File: data/json/items/gunmod/rail.json [hunk 5]
```diff
@@ -64,6 +82,60 @@
...
+    "mod_targets": [ "smg", "rifle", "shotgun", "crossbow", "launcher" ],
+    "min_skills": [ [ "gun", 1 ] ],
+    "add_mod": [ [ "sights", 1 ] ],
+    "flags": [ "DISABLE_SIGHTS" ]
+  },
+  {
+    "id": "large_rail",
+    "type": "GUNMOD",
+    "name": "large sight rail",
+    "description": "A large sized rail, for attaching scopes and the like. This one fits large sized sights."
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/rail.json]:
```suggestion
    "description": "A large sized rail, for attaching scopes and the like.  This one fits large sized sights."
```

**Comment by hyper2snyper** [comment]
ok ive hit a wall, what can i do

**Comment by ZhilkinSerg** [comment]
Marking as stalled. Please ping me to reopen when you ready to pick this up where you left.

**Pull request closed without merging**